lib/internal/streams/add-abort-signal.js

```diff
@@ -0,0 +1,42 @@
+'use strict';
+
+const {
+  hideStackFrames,
+  codes
+} = require('internal/errors');
+
+const { ERR_INVALID_ARG_TYPE } = codes;
+
+const {
+  validateAbortSignal,
+} = require('internal/validators');
+
+let DOMException;
+const lazyDOMException = hideStackFrames((message, name) => {
+  if (DOMException === undefined)
+    DOMException = internalBinding('messaging').DOMException;
+  return new DOMException(message, name);
```

Comment:
This is going to be _very_ complicated for readable-stream. Could we avoid this?

Reply:
picked `AbortError` since [that's what the DOM suggests](https://dom.spec.whatwg.org/#abortcontroller-api-integration). However:

> Any web platform API using promises to represent operations that can be aborted must adhere to the following:

This is not a web platform API and we're not using promises here. So we really don't have to do this. 

---

That said: Why is this going to be complicated for ReadableStream? 

In browsers - this is quite easy and is just "new DOMException('The operation was aborted', 'AbortError')"

In Node - it's pretty easy to get an AbortError constructor from Node itself (for example by making an AbortError and then using .constructor)

Reply:
I don't want this code using any `internal` stuff. This depends on `DomException` and `internalBinding`. Those should not be part of streams. I already have the Node error system done. Please use a normal error.

Reply:
I'll concede and change it to a regular error.